lib/tbot/ssh_proxy.go

```diff
@@ -220,38 +221,61 @@ func resolveTargetHost(ctx context.Context, cfg client.Config, search, query str
 // resolveTargetHostWithClient resolves the target host using the provided
 // client and search and query parameters.
 func resolveTargetHostWithClient(
-	ctx context.Context, clt client.ListUnifiedResourcesClient, search, query string,
+	ctx context.Context, clt *client.Client, search, query string,
 ) (types.Server, error) {
-	resources, _, err := client.GetUnifiedResourcePage(ctx, clt, &proto.ListUnifiedResourcesRequest{
-		// We only want a single node, but, we set limit=2 so we can throw a
-		// helpful error when multiple match. In the happy path, where a single
-		// node matches, this does not degrade performance because even if
-		// limit=1 the UnifiedResource cache will still iterate to the end to
-		// determine if there is a NextKey to return.
-		Limit:               2,
-		Kinds:               []string{types.KindNode},
+	resp, err := clt.ResolveSSHTarget(ctx, &proto.ResolveSSHTargetRequest{
 		SearchKeywords:      libclient.ParseSearchKeywords(search, ','),
 		PredicateExpression: query,
-		SortBy:              types.SortBy{Field: types.ResourceKind},
 	})
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-	if len(resources) == 0 {
-		return nil, trace.NotFound("no matching SSH hosts found for search terms or query expression")
-	}
-	if len(resources) > 1 {
-		names := make([]string, len(resources))
-		for i, res := range resources {
-			names[i] = res.GetName()
+	switch {
+	//TODO(tross): DELETE IN v20.0.0
+	case trace.IsNotImplemented(err):
+		resources, err := client.GetAllUnifiedResources(ctx, clt, &proto.ListUnifiedResourcesRequest{
```

Comment:
If you move this section back into its own method it'll just be a renaming of the old method.